enterprise/cmd/frontend/internal/auth/saml/middleware.go

```diff
@@ -136,27 +136,31 @@ func samlSPHandler(db database.DB) func(w http.ResponseWriter, r *http.Request)
 		case "/acs":
 			info, err := readAuthnResponse(p, r.FormValue("SAMLResponse"))
 			if err != nil {
-				log15.Error("Error validating SAML assertions. Set the env var INSECURE_SAML_LOG_TRACES=1 to log all SAML requests and responses.", "err", err)
+				logger.Error("Error validating SAML assertions. Set the env var INSECURE_SAML_LOG_TRACES=1 to log all SAML requests and responses.", log.Error(err))
 				http.Error(w, "Error validating SAML assertions. Try signing in again. If the problem persists, a site admin must check the configuration.", http.StatusForbidden)
 				return
 			}
 
 			if !allowSignin(p, info.groups) {
-				log15.Warn("Error authorizing SAML-authenticated user.", "AccountID", info.spec.AccountID, "Expected groups", p.config.AllowGroups, "Got", info.groups)
+				groups := make([]string, len(info.groups))
```

Comment:
```suggestion
				groups := make([]string, 0, len(info.groups))
```

This is incorrect, it would create groups like this: 

```go
groups := []string{"", "", "", ..., info.groups[0], info.groups[1], ...}
```

Reply:
Good catch!